.github/workflows/test.yml

```diff
@@ -159,11 +159,36 @@ jobs:
         run: |
           set -o pipefail
           diff-cover --compare-branch=${{ env.compare-branch }} --fail-under=100 --html-report=coverage-reports/diff-cover.html --markdown-report=coverage-reports/diff-cover.md coverage-reports/coverage.xml | tee coverage-reports/diff-cover-stdout
+          COV_STATUS="${PIPESTATUS[0]}"
+          echo "COV_STATUS=$COV_STATUS" >> "$GITHUB_ENV"
+
+      - name: Add diff coverage report comment to PR
+        if: env.COV_STATUS != '0'
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          # Transform diff-cover output into a markdown table
+          awk 'BEGIN{print "| File | Coverage | Missing Lines |"; print "|------|----------|---------------|"}
+              /.*\.py/ {
+                split($0,a,"(");
+                gsub(")","",a[2])
+                if (index(a[2], ":") == 0) {
+                  print "| " a[1] " | " a[2] " | None |"
+                } else {
+                  split(a[2],b,":");
+                  print "| " a[1] " | " b[1] " | " substr(b[2], 15) " |"
+                }
+              }' coverage-reports/diff-cover-stdout > coverage-reports/diff-cover-md-table
+          cat coverage-reports/diff-cover-md-table >> "$GITHUB_STEP_SUMMARY"
+          # Add comment to PR
+          PR_NUM=$(jq -r '.number' "$GITHUB_EVENT_PATH")
+          gh pr edit "$PR_NUM" --add-label "coverage-diff"
+          gh pr comment "$PR_NUM" -F coverage-reports/diff-cover-md-table
```

Comment:
perhaps it's asking for a lot, but it would be nice to update the existing comment, if there is one

Reply:
Done. It actually embeds a hidden comment ID when it adds the comment, and removes that comment before posting a new one, or removes it up on success change.